### DISCUSSION TITLE
[1 febbraio 2022 - B] Esercizio 2

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/psd2223#212] [Esami]
**Esercizio 2 (1+1+1+2+2+1 punti)**

La funzione di 4 variabili, 𝑓(a,b,c,d), vale 0:

- se $(a + b̅ )(a̅ + b)=1$
- se $a \oplus b \oplus d=0$

---

- Si stenda la tavola di verità di _f_
- Si scrivano le espressioni canoniche SOP e POS di _f_
- Si scrivano le espressioni minimali SOP e POS di _f_
- Si realizzi la funzione _f_ utilizzando un MUX ad 4 ingressi
- Si scriva l’espressione di _f_ in forma ALL-NAND

**Comment by Elia-Belli** [comment]
- Stendo la tavola di verità mettendo 0 nelle linee indentificate dalle 2 condizioni, altrimenti 1

- $Y = \bar a b \bar c \bar d + \bar a b c \bar d + a \bar b \bar c \bar d + a \bar b c \bar d$
  $Y = (a+b+c+d)(a+b+c+\bar d)(a+b+\bar c+d)(a+b+\bar c+\bar d)(a+\bar b+c+\bar d)(a+\bar b+\bar c+\bar d)(\bar a+b+c+\bar d)$
$(\bar a+b+\bar c+\bar d)(\bar a+\bar b+c+d)(\bar a+\bar b+c+\bar d)(\bar a+\bar b+\bar c+d)(\bar a+\bar b+\bar c+\bar d)$

- In rosso POS, in blu SOP
$Y = \bar d (a+b)(\bar a + \bar b)$
$Y = \bar a b \bar d + a \bar b \bar d$

- $Y = \overline {(\overline {\bar a b \bar d})(\overline {a \bar b \bar d})}$
 ![t](https://user-images.githubusercontent.com/107944476/212079698-0c583249-a256-4c28-9693-111803403924.png)

**Comment by FeddyLix17** [comment]
tutto corretto

**Comment by fraxb04** [comment]
@Elia-Belli 
in questo caso però non basta negare una volta tutta la SOP?
In questo modo otterremo subito tutti NAND... oppure per full-NAND si intende che devono essere tutti gli AND negati e anche tutta la funzione negata?

**Comment by Elia-Belli** [comment]
> @Elia-Belli in questo caso però non basta negare una volta tutta la SOP? In questo modo otterremo subito tutti NAND... oppure per full-NAND si intende che devono essere tutti gli AND negati e anche tutta la funzione negata?

Non puoi negare una volta sola la $SOP$ altrimenti modifichi la funzione, mentre negare due volte la fa rimanere invariata. Comunque, appunto come dice il nome $ALL-NAND$, devono essere tutte porte $NAND$, cioè tutti $AND$ negati.

**Comment by PierluigiCovone98** [comment]
Per quanto riguarda l'espressione di _f_ in forma ALL-NAND:
Anche le porte NOT devono essere scritte come porte NAND. In particolare, per un generico letterale **X** si ha: **X' = (XX)'**.

**Reply by Elia-Belli** [reply]
Corretto, ma per "pulizia" lo omettiamo
